storage/replica.go

```diff
@@ -3177,6 +3177,12 @@ func (r *Replica) executeBatch(
 		optimizePuts(batch, ba.Requests, ba.Header.DistinctSpans)
 	}
 
+	// Update the node clock with the serviced request. This maintains a high
+	// water mark for all ops serviced, so that received ops without a timestamp
+	// specified are guaranteed one higher than any op already executed for
+	// overlapping keys.
+	r.store.Clock().Update(ba.Header.Timestamp)
```

Comment:
Shouldn't have to do much with the clock update code. I could take a look, but I'm trying to clear out my queue and avoid distraction, so if anyone else is free I'd prefer if they took it.